src/silly_sprites/opengl/buffer_object.cpp

```diff
@@ -0,0 +1,97 @@
+#include "buffer_object.hpp"
+
+namespace sly::gl {
+    void BufferObject::init() {
+        generate_vertex_array();
+        generate_vertex_buffer();
+        generate_element_buffer();
+
+        glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), static_cast<void*>(0));
+        glEnableVertexAttribArray(0);
+    }
+
+    void BufferObject::generate_vertex_array() {
+        glGenVertexArrays(1, &m_vao);
+        bind_vertex_array();
+    }
+
+    void BufferObject::generate_vertex_buffer() {
+        glGenBuffers(1, &m_vbo);
+        bind_vertex_buffer();
+        data_vertex_buffer();
+    }
+
+    void BufferObject::generate_element_buffer() {
+        glGenBuffers(1, &m_ebo);
+        bind_element_buffer();
+        data_element_buffer();
+    }
+
+    void BufferObject::bind_vertex_array() const {
+        glBindVertexArray(m_vao);
+    }
+
+    void BufferObject::bind_vertex_buffer() const {
+        glBindBuffer(GL_ARRAY_BUFFER, m_vbo);
+    }
+
+    void BufferObject::bind_element_buffer() const {
+        glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_ebo);
+    }
+
+    void BufferObject::data_vertex_buffer() const {
+        glBufferData(GL_ARRAY_BUFFER, static_cast<GLsizeiptr>(sizeof(float) * m_points.size()), m_points.data(), GL_STATIC_DRAW);
```

Comment:
I think the warning/error there is correct (because we cast from 64 bit unsigned to long int, which is signed) and we need the cast. But in general we should use GSL for narrowing conversions:
```suggestion
        glBufferData(GL_ARRAY_BUFFER, gsl::narrow_cast<GLsizeiptr>(sizeof(float) * m_points.size()), m_points.data(), GL_STATIC_DRAW);
```